hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAclRequest.java

```diff
@@ -146,6 +153,36 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
    */
   abstract String getPath();
 
+  public BucketLayout getBucketLayout(OzoneManager ozoneManager) {
+    BucketLayout bucketLayout = BucketLayout.LEGACY;
+    OmBucketInfo buckInfo = null;
+    try {
+      ObjectParser objectParser = new ObjectParser(getPath(),
+          OzoneManagerProtocolProtos.OzoneObj.ObjectType.KEY);
+
+      String volume = objectParser.getVolume();
+      String bucket = objectParser.getBucket();
+
+      String buckKey =
+          ozoneManager.getMetadataManager().getBucketKey(volume, bucket);
+
+      try {
+        buckInfo =
+            ozoneManager.getMetadataManager().getBucketTable().get(buckKey);
+        if (buckInfo == null) {
+          LOG.error("Bucket not found: {}/{} ", volume, bucket);
+          return BucketLayout.LEGACY;
+        }
+        bucketLayout = buckInfo.getBucketLayout();
+      } catch (IOException e) {
+        LOG.debug("Failed to get the value for the key: " + buckKey);
```

Comment:
Please change it to` LOG.error` 

`LOG.error("Failed to get bucket for the key: {}", buckKey);`

Reply:
Done